enhancements/custom-alert-configuration.md

```diff
@@ -0,0 +1,438 @@
+to exclude PodDisruptionBudgets owned by KubeVirt from
+`PodDisruptionBudgetAtLimit` alert. So the following label would have to be
+placed on these PDBs:
+`alerts.openshift.io/PodDisruptionBudgetAtLimit: excluded`.
+The owner of the alert would have to make sure to exclude these objects in the
+alert query.
+
+Nevertheless, only allowed actors should have the power to set these labels to
+prevent misuse.
+
+In the case of the `PodDisruptionBudgetAtLimit` alert, excluding other users'
+PDBs from the alerting would have a detrimental effect on the cluster
+stability. Users could set this on any PDB and admins would have no reliable
+way of knowing whether something is blocking the node maintenance or an
+upgrade.
+
+We can use an alternative admission control based on [CEL][5] expressions that
+does not have the maintenance and reliability issues of an admission webhook.
+We can create [ValidatingAdmissionPolicy][6] for PDB objects to prevent a misuse
+of these labels. The validation expression would test for a presence of a
+specified label in the PDB object and would check for the Kubevirt user/SA
+in the request to decide if it is allowed.
+
+We will discuss introduction of a CRD in [API Extensions](#api-extensions)
+to support customization of allowed users.
+
+
+#### Variation [optional]
+
+
+### API Extensions
+
+Since we would like to parametrize users or service accounts, it is difficult
+to use built-in objects like ConfigMap and Secret. This is because they do not
+allow special characters like `:` in the data key which the service accounts
+have. And it seems there is no way to lookup a value in a map at this moment.
+
+To support this, we suggest a following CustomResourceDefinition:
+
+```go
+
+//
+// AlertExclusionConfig describes the alert exclusion configuration
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +genclient
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type AlertExclusionConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec AlertExclusionConfigSpec `json:"spec"`
+}
+
+// AlertExclusionConfigSpec describes the alert exclusion configuration
+type AlertExclusionConfigSpec struct {
+    // subjects to exclude from the alert
+    //
+    // +optional
+    Subjects map[string]string `json:"subjects,omitempty"
+}
+
+```
+
+The owner of the alert would create the AlertExclusionConfig resource and
+manage the subjects. In the case of KubeVirt, this would include determining
+if and in what namespace KubeVirt is installed.
+
+```yaml
+kind: AlertExclusionConfig
+apiVersion: config.openshift.io/v1
+metadata:
+  name: pod-disruption-budget-at-limit
+spec:
+  subjects:
```

Comment:
we should have the ability to match on pod names (launcher in KubeVirt's case).

Reply:
It could be one of the options if scoped only to KubeVirt, but we are trying to come up with a general approach that could be used by other components as well.